lib/leaf/match.js

```diff
@@ -0,0 +1,32 @@
+module.exports = function( property, value, extra ) {
+  if( !property || !value) {
+    return null;
+  }
+
+  const query = {
+    match: {
+      [property]: {
+        query: value
+      }
+    }
+  };
+
+  const extra_params = ['boost',
+    'operator',
+    'analyzer',
+    'cutoff_frequency',
+    'fuzziness',
+    'max_expansions',
+    'prefix_length',
+    'fuzzy_transpositions',
+    'minimum_should_match',
+    'zero_terms_query'];
+
+  extra_params.forEach(function(param) {
```

Comment:
sorry for sounding pedantic, could we do the `if(extra)` check before the iteration, in the most simple use case this would perform 10 iterations unnecessarily.

it's not really a huge deal, but it's nice to try to optimize performance on primitive building blocks like this because the effects will propagate up and multiply.

Reply:
could use fat arrow function syntax here for succinctness?

Reply:
Same as above: I bet V8 can do this for us, and wrapping the entire loop in another if statement  will make it that much harder to read. I'd rather not change it.